infra/modules/aws-psoxy-instance/main.tf

```
@@ -0,0 +1,197 @@
+        "Sid": ""
+      }
+    ]
+  })
+}
+
+resource "aws_lambda_function" "psoxy-instance" {
+  function_name    = var.function_name
+  role             = aws_iam_role.iam_for_lambda.arn
+  handler          = "co.worklytics.psoxy.Handler"
+  runtime          = "java11"
+  filename         = var.path_to_function_zip
+  source_code_hash = filebase64sha256(var.path_to_function_zip)
+  timeout          = 55 # seconds
+  memory_size      = 512 # megabytes
+
+  environment {
+    # NOTE: can use merge() to combine var map from config with additional values
+    variables = yamldecode(file(var.path_to_config))
+  }
+}
+
+
+// User: arn:aws:sts::874171213677:assumed-role/iam_for_lambda/psoxy-gdirectory is not authorized to perform:  on resource: arn:aws:ssm:us-east-1:874171213677:parameter/RULES because no identity-based policy allows the ssm:GetParameter action
+
+resource "aws_iam_policy" "policy" {
+  name        = "${var.function_name}_ssmGetParameters"
+  description = "Allow lambda function role to read SSM parameters"
+
+  policy = jsonencode(
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": [
+        "ssm:GetParameter*"
+      ],
+      "Effect": "Allow",
+      "Resource": "*"
+      # TODO: limit to SSM parameters in question
+      # "Resource": "arn:aws:ssm:us-east-2:123456789012:parameter/prod-*"
+    }
+  ]
+})
+
+}
+
+
+
+resource "aws_iam_role_policy_attachment" "basic" {
+  role       = aws_iam_role.iam_for_lambda.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
+}
+
+resource "aws_iam_role_policy_attachment" "policy"{
+  role       = aws_iam_role.iam_for_lambda.name
+  policy_arn = aws_iam_policy.policy.arn
+}
+
+
+resource "local_file" "todo" {
+  filename = "TODO - deploy ${var.function_name}.md"
+  content  = <<EOT
+First, from `java/core/` within a checkout of the Psoxy repo, package the core proxy library:
+
+```shell
+cd ../../java/core
+mvn package install
+```
+
+Second, from `java/impl/aws` within a checkout of the Psoxy repo, package an executable JAR for the
+cloud function with the following command:
+
+```shell
+cd ../../java/impl/aws
+mvn package
+```
+
+Third, run the following deployment command from `java/impl/aws` folder within your checkout:
+
+```shell
+aws sts assume-role --duration 900 --role-arn ${var.aws_assume_role_arn} --role-session-name deploy-lambda > temporal-credentials.json
+export CALLER_ACCESS_KEY_ID=`cat temporal-credentials.json| jq -r '.Credentials.AccessKeyId'`
+export CALLER_SECRET_ACCESS_KEY=`cat temporal-credentials.json| jq -r '.Credentials.SecretAccessKey'`
+export CALLER_SESSION_TOKEN=`cat temporal-credentials.json| jq -r '.Credentials.SessionToken'`
+rm temporal-credentials.json
+aws sts get-caller-identity
+
+aws lambda update-function-code --function-name ${var.function_name} --zip-file fileb://target/psoxy-aws-1.0-SNAPSHOT.jar
+
+unset CALLER_ACCESS_KEY_ID CALLER_SECRET_ACCESS_KEY CALLER_SESSION_TOKEN
```

oops, this is wrong... my bad. These need to be 
```suggestion
export AWS_ACCESS_KEY_ID=`cat temporal-credentials.json| jq -r '.Credentials.AccessKeyId'`
export AWS_SECRET_ACCESS_KEY=`cat temporal-credentials.json| jq -r '.Credentials.SecretAccessKey'`
export AWS_SESSION_TOKEN=`cat temporal-credentials.json| jq -r '.Credentials.SessionToken'`
rm temporal-credentials.json
aws sts get-caller-identity

aws lambda update-function-code --function-name ${var.function_name} --zip-file fileb://target/psoxy-aws-1.0-SNAPSHOT.jar

unset AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY AWS_SESSION_TOKEN
```
The deploy does use the AWS env variables to work.
I renamed that to CALLER_X when testing because there is no real reason to overwrite the environment (as passing it as env variables didn't work in `awscurl`)